addons/purchase_stock/models/stock.py

```diff
@@ -163,10 +164,84 @@ def _prepare_move_default_values(self, return_line, new_picking):
 class Orderpoint(models.Model):
     _inherit = "stock.warehouse.orderpoint"
 
+    show_supplier = fields.Boolean('Show supplier column', compute='_compute_show_suppplier')
+    supplier_id = fields.Many2one(
+        'product.supplierinfo', string='Vendor',
+        domain="['|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
```

Comment:
missing companies check

Reply:
check_company isn't enough, you have to adapt the domain since it is redefined :(